doc/specs/#182 - Support for installation of portable standalone apps.md

```diff
@@ -0,0 +1,213 @@
+
+Using Gitlab.gitlab runner as an example would result in the .exe being placed in "%LOCALAPPDATA%/Microsoft/WinGet/Packages/Gitlab.gitlab-runner/"
+If a user has configured a path to be used for portable applications, that path should be honored. The user should also be able to specify if they want all portable packages placed in the same directory, or in a directory per package.
+
+A related setting for the uninstall scenario will be able to specify the default behavior for either "--clean" or "--purge".
+
+### Install
+
+When the portable application is being installed by the Windows Package Manager, an entry will be created in Windows Apps & Features so the user will be able to see that the application is installed.
+
+The "AppPath" registry subkey will identify the path to the application so the user will be able to execute the program from any path via command prompt.
+
+The data from “AppsAndFeaturesEntry” in the manifest will be used to specify the name of the package in Windows Apps & Features.  Constraints for manifests with portable packages are covered below in the section on validating manifests. If the “AppsAndFeaturesEntry” doesn’t have a “DisplayName” value, then we will use the “PackageName”.
+The Product Code will be populated with the "PackageIdentifier" if it is not specified in the “AppsAndFeaturesEntry”.
+
+No shortcuts or icons will be created for the package.
+
+>Note: Some portable programs may require dependencies. Dependencies are not covered in this specification and are not yet supported as of the writing of this specification.
+
+#### Scope Behavior
+
+By default, portable apps will be installed with the "User" scope unless specified otherwise by the user through the "--scope" argument or in their settings. This also means that the "scope" field in the manifest will be ignored when installing portable apps. Since portable apps are unique in that they are simply standalone executables that are copied to a specific install location based on scope, it does not seem reasonable to have the manifest enforce an install behavior that is not required. Manifest validation would also need to be updated to show an error to the user that the "scope" field does not apply if a portable installer type is specified.
+
+#### Installation from multiple sources:
+
+If the user chooses to install the same package but from a secondary source, the Windows Package Manager will create an additional subdirectory to indicate the source that the package came from. Since entries in the registry must be unique, if the entry in the "App Paths" registry conflicts with an existing entry, then the user will be notified of the conflict and the subkey value will be updated to reflect the change in executable path. In other words, the latest installed package will take precedence. This will require utilizing the installed packages data store to determine whether a package is being installed from two different sources.
```

Comment:
I don't think that an additional subdirectory should be created; it makes it harder to visually scan the packages directory to find things.  Something more like `PackageIdentifier_Source2` would make it easier to find both the first and second package directories together.

Reply:
Additionally, I'm not sure that we should do "last writer wins (LWW)" semantics.  I think that is appropriate if we did nothing else, but I would also like to have a command like `winget app-path apply PackageFoo`.  If we had that command, I would think it better to have "first writer wins (FWW)" and allow the user to use the new command to apply the appropriate app-path.

We could start with FWW and an additional parameter to specify LWW for now, then add the app path update command in the future.

Reply:
I updated the description to reflect your suggestion. I also agree that not having a separate directory just for source would make it easier to find both packages in the same directory. 

I included further clarification for having FWW be our default behavior as well as having an argument to support allowing users to have a LWW behavior if they want that using the --lww argument